api/index.js

```diff
@@ -42,18 +91,39 @@ app.get('/config', async (req, res) => {
   }
 });
 
-app.post('/attachment', async(req, res) => {
-  var userToken = req.headers[USER_TOKEN_HEADER_KEY];
-  if (!userToken) {
-    return res.status(HttpStatus.BAD_REQUEST).json({error: CONTAINER_NAME + ` request header is missing`});
+
+app.use(async (req, res, next) => {
+  try{
+    // Verify User Account header:
+    var userAccount = req.headers[USER_ACCOUNT_HEADER_KEY];
+    if (!userAccount) return res.status(HttpStatus.BAD_REQUEST).json({error: USER_ACCOUNT_HEADER_KEY + ` request header is missing`});
+
+    // Authentication request JWT:
+    var authHeader = req.headers['authorization'] || '';
+    var elements = authHeader.split(' ');
+    if (elements.length !== 2) return res.status(HttpStatus.BAD_REQUEST).json({error: `invalid authorization token`});
+    var requestJwt = elements[1];
+    if(!verifyJwt(requestJwt)) return res.status(HttpStatus.BAD_REQUEST).json({error: `Request is unauthorised with this authorization token`});
+    console.log(`request authenticated.`);
+
+    var userExchangeId = parseAppctx(requestJwt).msexchuid;
+    req.userInfo = {userAccount, userExchangeId};
```

Comment:
Please change to 
req.userInfo = {account, exchangeId};

since both account and exchangeId are under the userInfo object, it's clear that they are the user's account and exchange id, no need to use them in the parameter name

Reply:
done